tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -740,175 +843,181 @@ public void testDescribeUnavailablePartitions(String quorum) throws ExecutionExc
-            () -> {
-                try {
-                    return !adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get()
-                        .get(tp).addingReplicas().isEmpty();
-                } catch (InterruptedException | ExecutionException e) {
-                    throw new RuntimeException(e);
-                }
-            },
-            () -> "Reassignment didn't add the second node",
-            org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS, 100L);
-
-        ensureConsistentKRaftMetadata();
-
-        // describe the topic and test if it's under-replicated
-        String simpleDescribeOutput = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--topic", testTopicName));
-        String[] simpleDescribeOutputRows = simpleDescribeOutput.split(System.lineSeparator());
-        assertTrue(simpleDescribeOutputRows[0].startsWith(String.format("Topic: %s", testTopicName)),
-                "Unexpected describe output: " + simpleDescribeOutputRows[0]);
-        assertEquals(2, simpleDescribeOutputRows.length,
-                "Unexpected describe output length: " + simpleDescribeOutputRows.length);
-
-        String underReplicatedOutput = captureDescribeTopicStandardOut(buildTopicCommandOptionsWithBootstrap("--describe", "--under-replicated-partitions"));
-        assertEquals("", underReplicatedOutput,
-            String.format("--under-replicated-partitions shouldn't return anything: '%s'", underReplicatedOutput));
-
-        int maxRetries = 20;
-        long pause = 100L;
-        long waitTimeMs = maxRetries * pause;
-        AtomicReference<PartitionReassignment> reassignmentsRef = new AtomicReference<>();
-
-        TestUtils.waitUntilTrue(() -> {
-            try {
-                PartitionReassignment tempReassignments = adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get().get(tp);
-                reassignmentsRef.set(tempReassignments);
-            } catch (InterruptedException | ExecutionException e) {
-                throw new RuntimeException("Error while fetching reassignments", e);
-            }
-            return reassignmentsRef.get() != null;
-        }, () -> "Reassignments did not become non-null within the specified time", waitTimeMs, pause);
+            adminClient.alterPartitionReassignments(Collections.singletonMap(tp,
+                    Optional.of(new NewPartitionReassignment(Collections.singletonList(targetReplica))))).all().get();
+
+            // let's wait until the LAIR is propagated
+            TestUtils.waitForCondition(
+                    () -> {
+                        try {
+                            return !adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get()
+                                    .get(tp).addingReplicas().isEmpty();
+                        } catch (InterruptedException | ExecutionException e) {
```

Comment:
we don't need this try-catch, right?

Reply:
this can be 
```
TestUtils.waitForCondition(
                    () -> !adminClient.listPartitionReassignments(Collections.singleton(tp)).reassignments().get()
                        .get(tp).addingReplicas().isEmpty(), 60000, "Reassignment didn't add the second node"
            );
            ```
no need to wrap it in try and catch.